python/python/glide/async_commands/cluster_commands.py

```diff
@@ -514,3 +516,52 @@ async def flushall(
             TClusterResponse[TOK],
             await self._execute_command(RequestType.FlushAll, args, route),
         )
+
+    async def cluster_scan(
+        self,
+        cursor: ClusterScanCursor,
```

Comment:
For user this should be just a string (or an arbitrary Object)
```suggestion
        cursor: str,
```
We should not export this type to user API too

Reply:
We have to hold it as an object since we need to trigger delete from the table when it is dropped.

Reply:
Deserialize the string into an object and do whatever you need.

Reply:
I don't understand why. We need a lifetime of an object when it is in the user side so we know to trigger del when he drop it.
If the user get a str we cant know when to drop it. the lifetime of a scan is not based on finishing the scan, but based on the user holding the cursor, which cant be implemented on a str.

Reply:
I don't follow. By the time the user gets this, the data has already been deserialized so there's no need to extend the life time of the data in the Rust layer. Similarly in a native Rust scenario, when the user calls this function, they'll have a reference to the data so its lifetime should be as long as they need it (no need for the library to store the data).